### PR TITLE
chore(deps): update dependency google.cloud.spanner.data to v5.0.0-beta04

### PR DESCRIPTION
**Pull request by ketanv3**
This PR upgrades google.cloud.spanner.data to v5.0.0-beta04 dependency and fixes the broken samples.

### CODE REVIEW

File: spanner/api/Spanner.Samples/BatchReadRecordsAsync.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ public class BatchReadRecordsAsyncSample
         using var connection = new SpannerConnection(connectionString);
         await connection.OpenAsync();
 
-        using var transaction = await connection.BeginReadOnlyTransactionAsync();
+        using var transaction = await connection.BeginTransactionAsync(SpannerTransactionCreationOptions.ReadOnly, cancellationToken: default);
```
Comment (amanda-tarafa) [inline spanner/api/Spanner.Samples/BatchReadRecordsAsync.cs:33]: This one needs to be:

```suggestion
        using var transaction = await connection.BeginTransactionAsync(
            SpannerTransactionCreationOptions.ReadOnly.WithIsDetached(true), cancellationToken: default);
```

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
All good, except for one change that needed to go further. Thanks for taking the time to do this.

File: spanner/api/Spanner.Samples/BatchReadRecordsAsync.cs (hunk 1 above)
Comment (ketanv3) [reply spanner/api/Spanner.Samples/BatchReadRecordsAsync.cs]: Updated!

**Comment by amanda-tarafa** [comment]
Rerunning tests to confirm everything is working, I'll stamp tomorrow. Thanks!

**Review by amanda-tarafa — APPROVED** [review]
Thanks!

**Pull request merged**